src/library/trackset/baseplaylistfeature.cpp

```diff
@@ -600,6 +600,101 @@ void BasePlaylistFeature::slotExportPlaylist() {
     }
 }
 
+void BasePlaylistFeature::slotExportPlaylists() {
+    const PlaylistDAO::HiddenType hidden = PlaylistDAO::PLHT_NOT_HIDDEN;
+
+    QList<QPair<int, QString>> allPlaylists = m_playlistDao.getPlaylists(hidden);
+
+    QPair<int, QString> thePair;
+    int playlistId;
+    QString playlistName;
+
+    for (int i = 0; i < allPlaylists.length(); i++) {
+        thePair = allPlaylists.at(i);
+        playlistId = thePair.first;
+        playlistName = m_playlistDao.getPlaylistName(playlistId);
+
+        if (playlistId == kInvalidPlaylistId) {
+            return;
+        }
+
+        // replace separator character with something generic
+        playlistName = playlistName.replace(QDir::separator(), kUnsafeFilenameReplacement);
```

Comment:
Please consider which part of this body can be extracted into a common function called from slotExportPlaylists() and slotExportPlaylist()

Reply:
Completed. Let me know what u think